xdsl/backend/riscv/register_allocation.py

```diff
@@ -24,6 +25,95 @@ def allocate_registers(self, module: ModuleOp) -> None:
         raise NotImplementedError()
 
 
+class RegisterAllocatorLivenessBlockNaive(RegisterAllocator):
```

Comment:
Can you please document this class with a detailed but high-level description of what its doing? The code is a bit hard to follow, would be nice to have a strategy outline

Reply:
I added a docstring, but this is a textbook algorithm and rather simplified.